anomalib/data/visa.py

```diff
@@ -0,0 +1,245 @@
+"""Visual Anomaly (VisA) Dataset (CC BY-NC-SA 4.0).
+
+Description:
+    This script contains PyTorch Dataset, Dataloader and PyTorch
+        Lightning DataModule for the Visual Anomal (VisA) dataset.
+    If the dataset is not on the file system, the script downloads and
+        extracts the dataset and create PyTorch data objects.
+License:
+    The VisA dataset is released under the Creative Commons
+    Attribution-NonCommercial-ShareAlike 4.0 International License
+    (CC BY-NC-SA 4.0)(https://creativecommons.org/licenses/by-nc-sa/4.0/).
+Reference:
+    - Zou, Y., Jeong, J., Pemula, L., Zhang, D., & Dabeer, O. (2022). SPot-the-Difference
+      Self-supervised Pre-training for Anomaly Detection and Segmentation. In European
+      Conference on Computer Vision (pp. 392-408). Springer, Cham.
+"""
+
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+# Subset splitting code adapted from https://github.com/amazon-science/spot-diff
+# Original licence: Apache-2.0
+
+import csv
+import logging
+import shutil
+from pathlib import Path
+from typing import Optional, Tuple, Union
+
+import albumentations as A
+import cv2
+
+from anomalib.data.base import AnomalibDataModule, AnomalibDataset
+from anomalib.data.task_type import TaskType
+from anomalib.data.utils import (
+    DownloadInfo,
+    InputNormalizationMethod,
+    Split,
+    TestSplitMode,
+    ValSplitMode,
+    download_and_extract,
+    get_transforms,
+)
+
+from .mvtec import make_mvtec_dataset
+
+logger = logging.getLogger(__name__)
+
+EXTENSIONS = [".png", ".jpg", ".JPG"]
+
+DOWNLOAD_INFO = DownloadInfo(
+    name="VisA",
+    url="https://amazon-visual-anomaly.s3.us-west-2.amazonaws.com/VisA_20220922.tar",
+    hash="ef908989b6dc701fc218f643c127a4de",
+)
+
+
+class VisaDataset(AnomalibDataset):
+    """VisA dataset class.
+
+    Args:
+        task (TaskType): Task type, ``classification``, ``detection`` or ``segmentation``
+        transform (A.Compose): Albumentations Compose object describing the transforms that are applied to the inputs.
+        split (Optional[Union[Split, str]]): Split of the dataset, usually Split.TRAIN or Split.TEST
+        root (Union[str, Path]): Path to the root of the dataset
```

Comment:
Same as my previous comment. Should we consider renaming `root` to `path`? I don't remember the motivation of using `root`